jonjoncardoso/data-science-workflow
# Initial docker structure

![image](https://user-images.githubusercontent.com/896254/128766165-175b6026-4484-4d3f-baa8-34dc3122428d.png)

# What this PR does

- Add a code directory structure that should be common enough for most exploratory data science projects
- Uses Docker to piece together all parts of the directory structure above
    - Files `notebooks/Dockerfile` and `notebooks/requirements.txt` control how Jupyter notebooks are  
- Adds the notebook [notebook-01-docker-structure.ipynb](https://github.com/jonjoncardoso/data-science-workflow/blob/feature/issue1/jon-initial-docker-structure/notebooks/notebook-01-docker-structure.ipynb)
- Closes #1 

# How to validate this PR

1. Make sure you have Git installed in your machine and [set up your Github account](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account). Alternatively: install the software [Gitkraken](https://www.gitkraken.com/) and connect it to your Github account.
2. Clone this repository to your computer
3. [Switch to the branch](https://www.freecodecamp.org/news/git-clone-branch-how-to-clone-a-specific-branch/) of this Pull Request. Its name is **/feature/issue1/jon-initial-docker-structure**
4. Follow through the setup instructions on the [README](https://github.com/jonjoncardoso/data-science-workflow/tree/feature/issue1/jon-initial-docker-structure#setup) to build and run the Docker container.
5. Read through the [notebook-01-docker-structure.ipynb](https://github.com/jonjoncardoso/data-science-workflow/blob/feature/issue1/jon-initial-docker-structure/notebooks/notebook-01-docker-structure.ipynb) on the browser by following the link or from within Jupyter. Is it clear enough? Does it achieve its objectives?
